src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
+    if (!this.isEmbedded_) {
```

Comment:
I don't know yet if they need to be treated differently or not, but it's very possible.

When we iframe an AMP document, the parent's URL is what's visible to the user - that's what they share. Thus, we need to update the parent window's location so that it's shared along with it. 

When an AMP is in a webview - the situation is different. AMP is basically in the standalone mode but talks to the container app. In this case, the user doesn't see address bar at all and so it's not clear which URL would actually end up being shared. It'd be good to research this a little bit.

A possible option here is to

1. Update window.location when AMP is NOT iframed (webview or not)
2. Message to viewer whenever viewer is available